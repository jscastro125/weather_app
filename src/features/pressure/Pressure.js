import React from 'react';
import { useSelector, useDispatch } from 'react-redux';

import {
    set,
    selectPressure,
} from './pressureSlice';
import Slider from "../../components/Slider/Slider";

export default function Pressure() {
    const pressure = useSelector(selectPressure);
    const dispatch = useDispatch();
    return (
        <Slider label="Pressure [ hPa ]" max={1030} min={970} value={pressure} onChange={(nPressure) => {
            dispatch(set(nPressure))
        }}/>
    );
}
