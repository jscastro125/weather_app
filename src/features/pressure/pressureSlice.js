import { createSlice } from '@reduxjs/toolkit';

export const pressureSlice = createSlice({
    name: 'pressure',
    initialState: {
        value: 1010,
    },
    reducers: {
        set: (state, action) => {
            state.value = action.payload;
        },
    },
});

export const { set } = pressureSlice.actions;

export const selectPressure = state => state.pressure.value;

export default pressureSlice.reducer;
