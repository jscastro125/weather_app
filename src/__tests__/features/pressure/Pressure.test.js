import React from "react";
import { create } from "react-test-renderer";
import { useSelector, useDispatch } from 'react-redux';

import Pressure from '../../../features/pressure/Pressure';
import * as pressureSlice from '../../../features/pressure/pressureSlice';

jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

describe("<Pressure />", () => {
    it("should match snapshot", () => {
        useSelector.mockImplementation(sel => sel());
        useDispatch.mockImplementation(() => () => {});
        const spyPressure = jest.spyOn(pressureSlice, 'selectPressure');
        spyPressure.mockReturnValue(30);
        const component = create(<Pressure />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
});
