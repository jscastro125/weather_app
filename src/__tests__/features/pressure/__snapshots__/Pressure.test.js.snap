// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pressure /> should match snapshot 1`] = `
<div
  className="sliderContainer"
>
  <div
    className="sliderLabel"
  >
    Pressure [ hPa ]
  </div>
  <input
    className="customSlider"
    max={1030}
    min={970}
    onChange={[Function]}
    type="range"
    value={30}
  />
  <div
    className="rangeValue"
  >
    <span
      style={
        Object {
          "left": "calc(-1566.6666666666667% + (313.33333333333337px))",
        }
      }
    >
      30
    </span>
  </div>
</div>
`;
